src/mon/NVMeofGwSerialize.h

```diff
@@ -590,6 +598,46 @@ inline void decode(
   DECODE_FINISH(bl);
 }
 
+inline void encode( const Gw_Epoch& gw_epoch,  ceph::bufferlist &bl)
+{
+  encode(gw_epoch.epoch, bl);
+}
+
+inline void encode(const std::map<NvmeGroupKey, GwEpoch>& gw_epoch,
+                   ceph::bufferlist &bl) {
```

Comment:
Do not create custom encoders for std containers -- as long as NvmeGroupKey and GwEpoch both have encoders there should be a std::map<NvmeGroupKep, GwEpoch> encoder defined via template in encoding.h (around line 525, grep for 'std::map').

Reply:
ok